### PR TITLE
OHRM5X-1661: Develop LDAP Config API

### PR DESCRIPTION
**Pull request by SandeepaOHRM**

### CODE REVIEW

File: src/plugins/orangehrmLDAPAuthenticationPlugin/Api/LDAPConfigAPI.php [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+     */
+    public function create(): EndpointResult
+    {
+        $dataMapping = $this->getRequestParams()->getArray(
+            RequestParams::PARAM_TYPE_BODY,
+            self::PARAMETER_DATA_MAPPING
+        );
+        $this->validate($dataMapping, $this->getParamRuleCollection());
+
+        $LDAPSettings = new LDAPSetting(
```
Comment (RajithaKumara) [inline src/plugins/orangehrmLDAPAuthenticationPlugin/Api/LDAPConfigAPI.php:109]:
```suggestion
        $ldapSettings = new LDAPSetting(
```
this also fine

File: src/plugins/orangehrmLDAPAuthenticationPlugin/Dto/LDAPSetting.php [hunk 2]
```diff
@@ -78,6 +106,27 @@ public static function fromString(string $string): self
...
+            'bindUserPassword' => $this->getBindUserPassword(),
+            'baseDN' => $this->getBaseDN(),
+            'searchScope' => $this->getSearchScope()
+        ]);
+    }
+
+    /**
+     * @return string
+     */
+    public function getEncodedAttributes(): string
```
Comment (RajithaKumara) [inline src/plugins/orangehrmLDAPAuthenticationPlugin/Dto/LDAPSetting.php:126]: not sure why we need this.
can't we use `__toString`

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: src/plugins/orangehrmLDAPAuthenticationPlugin/Api/LDAPConfigAPI.php (hunk 1 above)
Comment (SandeepaOHRM) [reply src/plugins/orangehrmLDAPAuthenticationPlugin/Api/LDAPConfigAPI.php:109]: done

File: src/plugins/orangehrmLDAPAuthenticationPlugin/Api/LDAPConfigAPI.php [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+     */
+    public function getValidationRuleForGetAll(): ParamRuleCollection
+    {
+        throw $this->getNotImplementedException();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function create(): EndpointResult
```
[the remark this answers is not in the document]
Comment (SandeepaOHRM) [reply src/plugins/orangehrmLDAPAuthenticationPlugin/Api/LDAPConfigAPI.php:101]: done

File: installer/Migration/V5_2_0/permission/api.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  description: 'API-v2 Admin   - LDAP Configurations'
+  api: OrangeHRM\LDAP\Api\LDAPConfigAPI
+  module: admin
+  allowed:
+    read: false
+    create: true
+    update: false
+    delete: false
+  permissions:
+    - { role: Admin, permission: { read: false, create: true, update: false, delete: false } }
```
[the remark this answers is not in the document]
Comment (SandeepaOHRM) [reply installer/Migration/V5_2_0/permission/api.yaml:15]: done

File: installer/Migration/V5_2_0/permission/api.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+apiv2_admin_ldap_config:
+  description: 'API-v2 Admin   - LDAP Configurations'
+  api: OrangeHRM\LDAP\Api\LDAPConfigAPI
+  module: admin
+  allowed:
+    read: false
+    create: true
+    update: false
+    delete: false
```
[the remark this answers is not in the document]
Comment (SandeepaOHRM) [reply installer/Migration/V5_2_0/permission/api.yaml:13]: done

File: installer/Migration/V5_2_0/permission/api.yaml [hunk 6]
```diff
@@ -0,0 +1,15 @@
+##########################
+#        LDAP
+##########################
+
+apiv2_admin_ldap_config:
+  description: 'API-v2 Admin   - LDAP Configurations'
```
[the remark this answers is not in the document]
Comment (SandeepaOHRM) [reply installer/Migration/V5_2_0/permission/api.yaml:6]: done

**Pull request merged**